mPower2/mPower2/PhoneRegistrationViewController.swift

```diff
@@ -81,8 +77,7 @@ class PhoneRegistrationViewController: RSDTableStepViewController {
             self.presentAlertWithYesNo(title: "Tester", message: "The phone number you entered is fictional. Do you want to sign in with a test ID?") { (tester) in
                 if tester {
                     // boop them over to the externalID sign-in step
-                    taskController.phoneSavedNextStepIdentifier = stepObject.nextStepIdentifier
-                    stepObject.nextStepIdentifier = "enterExternalId"
+                    self.assignSkipToIdentifier("enterExternalId")
```

Comment:
so just to be clear--when you navigate to the specified step, and then tap "back" to come back here, this assignment of a skip-to identifier will no longer be in effect (so that you can change your mind about using an external id and go back and sign in with a phone number)?

Reply:
Right.

The **result** (as oppose to the step) that is associated with a given step is replaced by a new instance every time the navigation goes to the step.

This is handled by the extension of `RSDTaskUIController` (so it applies to the task view controller) using this method:

```
    private func _move(to step: RSDStep, from previousStep: RSDStep?, direction: RSDStepDirection) {
        
        self.taskPath.currentStep = step
        let stepResult = step.instantiateStepResult()
        self.taskPath.appendStepHistory(with: stepResult)
        self.navigate(to: step, from: previousStep, direction: direction) { [weak self] (finished) in
            self?._finishMoving(to: step, from: previousStep, direction: direction)
        }
        
        self.hideLoadingIfNeeded()
    }
```

The step is asked to instantiate a fresh instance of the result. That is then appended to the task path, which also moves the previous result to a holding property on the task path that can be accessed by the step controller in order to populate with a previous result (if applicable).